geode-core/src/main/java/org/apache/geode/management/internal/operation/RegionOperationStateDistributionService.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.internal.operation;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+import java.util.function.Supplier;
+
+import org.apache.geode.annotations.VisibleForTesting;
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.RegionFactory;
+import org.apache.geode.cache.RegionShortcut;
+import org.apache.geode.internal.cache.InternalCache;
+import org.apache.geode.internal.cache.InternalRegionArguments;
+import org.apache.geode.internal.cache.RegionFactoryImpl;
+import org.apache.geode.management.api.ClusterManagementOperation;
+import org.apache.geode.management.runtime.OperationResult;
+
+public class RegionOperationStateDistributionService
+    implements OperationStateDistributionService {
+  private final Supplier<String> uniqueIdSupplier;
+  private final Region<String, OperationState<ClusterManagementOperation<OperationResult>, OperationResult>> region;
+
+  public static final String OPERATION_STATE_REGION_NAME = "OperationStateRegion";
```

Comment:
internal region needs to be started with "_" or "__".

Reply:
This has been changed to start with "__". I found a validator that does not allow names to start with "__".